src/main/java/org/jahia/modules/sam/healthcheck/probes/ModulesSpringUsageProbe.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * ==========================================================================================
+ * =                            JAHIA'S ENTERPRISE DISTRIBUTION                             =
+ * ==========================================================================================
+ *
+ *                                  http://www.jahia.com
+ *
+ * JAHIA'S ENTERPRISE DISTRIBUTIONS LICENSING - IMPORTANT INFORMATION
+ * ==========================================================================================
+ *
+ *     Copyright (C) 2002-2024 Jahia Solutions Group. All rights reserved.
+ *
+ *     This file is part of a Jahia's Enterprise Distribution.
+ *
+ *     Jahia's Enterprise Distributions must be used in accordance with the terms
+ *     contained in the Jahia Solutions Group Terms &amp; Conditions as well as
+ *     the Jahia Sustainable Enterprise License (JSEL).
+ *
+ *     For questions regarding licensing, support, production usage...
+ *     please contact our team at sales@jahia.com or go to http://www.jahia.com/license.
+ *
+ * ==========================================================================================
+ */
+package org.jahia.modules.sam.healthcheck.probes;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.gemini.blueprint.extender.support.ApplicationContextConfiguration;
+import org.jahia.modules.sam.Probe;
+import org.jahia.modules.sam.ProbeSeverity;
+import org.jahia.modules.sam.ProbeStatus;
+import org.jahia.osgi.BundleUtils;
+import org.jahia.osgi.FrameworkService;
+import org.osgi.framework.Bundle;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.MessageDigest;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Component(service = Probe.class, immediate = true)
+public class ModulesSpringUsageProbe implements Probe {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ModulesSpringUsageProbe.class);
+    private static final String EXCLUDE_JAHIA_MODULES_PROPERTY = "excludeJahiaModules";
+    private static final List<SpringUsageInfo> springUsages = new ArrayList<>();
+    private static String bundlesModificationCheksum = "";
+
+    private boolean excludeJahiaModules = true;
+
+    @Override
+    public String getName() {
+        return "ModulesSpringUsage";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Checks if some modules are using Spring on the Jahia instance";
+    }
+
+    @Override
+    public ProbeSeverity getDefaultSeverity() {
+        return ProbeSeverity.MEDIUM;
+    }
+
+    @Override
+    public ProbeStatus getStatus() {
+        this.searchForSpringUsageInBundles();
+        String configMessage = "(Jahia modules " + (excludeJahiaModules ? "not " : "") + "checked) ";
+        if (springUsages.isEmpty()) {
+            LOGGER.info(configMessage.concat("No modules using spring found "));
```

Comment:
Not sure about the logging in info level in the getStatus method.
Cloud is struggling with amount of log, and they are do request the healthcheck to avoid having the check the log.
I think we should move those log to be in debug log level.